database/src/pool/postgres.rs

```diff
@@ -342,7 +342,9 @@ static MIGRATIONS: &[&str] = &[
         CONSTRAINT job_queue_request_fk
             FOREIGN KEY (request_tag)
             REFERENCES benchmark_request(tag)
-            ON DELETE CASCADE
+            ON DELETE CASCADE,
+
+       CONSTRAINT job_queue_unique UNIQUE(request_tag, target, backend, profile)
```

Comment:
We also need to include the benchmark set here, otherwise we couldn't split the jobs across multiple collectors.

Reply:
An edge case we need to handle is a job having a collector id which is not in the collector configuration table.

I was thinking a foreign key however we'd then not be able to have a test for the `enqueue_benchmark_job(...)` without also inserting something into the `collector_config` table which would require creating more methods which we possibly won't be using. I think it needs a bit more thought so have left it out for now